src/adapter/src/coord.rs

```diff
@@ -1295,15 +1296,17 @@ impl Coordinator {
                         // MIR ⇒ MIR optimization (global)
                         let index_plan = optimize::Index::new(entry.name(), &idx.on, &idx.keys);
                         let global_mir_plan = optimizer.optimize(index_plan)?;
+
+                        // MIR ⇒ LIR lowering and LIR ⇒ LIR optimization (global)
+                        let mut global_lir_plan = optimizer.optimize(global_mir_plan.clone())?;
+
                         // Timestamp selection
                         let as_of = self.bootstrap_index_as_of(
-                            global_mir_plan.df_desc(),
-                            global_mir_plan.compute_instance_id(),
+                            global_lir_plan.df_desc(),
+                            idx.cluster_id,
                             idx.is_retained_metrics_object,
                         );
-                        let global_mir_plan = global_mir_plan.resolve(as_of);
-                        // MIR ⇒ LIR lowering and LIR ⇒ LIR optimization (global)
-                        let global_lir_plan = optimizer.optimize(global_mir_plan.clone())?;
+                        global_lir_plan.set_as_of(as_of);
```

Comment:
Is it necessary that the physical plan stored in `CatalogPlans` has an as-of set? If not we could do the timestamp selection even later, before we announce the dataflow to the compute controller.

Reply:
Not at the moment, and I don't have specific reasons to want to do it for now.